reorgdetector/reorgdetector_db.go

```diff
@@ -26,28 +26,23 @@ func (rd *ReorgDetector) getTrackedBlocks() (map[string]*headersList, error) {
 	}
 	currentID := headersWithID[0].SubscriberID
 	currentHeaders := []header{}
-	for i := 0; i < len(headersWithID); i++ {
-		if i == len(headersWithID)-1 {
-			currentHeaders = append(currentHeaders, header{
-				Num:  headersWithID[i].Num,
-				Hash: headersWithID[i].Hash,
-			})
+	for _, row := range headersWithID {
+		// If the subscriber ID changes, save the current group
```

Comment:
Just curious, how can subscriber ID change?

Reply:
I've took a look again to the code, and I understand it now. I will resolve the comment.